bazooka/bazooka.go

```diff
@@ -0,0 +1,351 @@
+package bazooka
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"strings"
+
+	"github.com/BOPR/config"
+	"github.com/BOPR/core"
+	"github.com/BOPR/log"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rpc"
+	tmLog "github.com/tendermint/tendermint/libs/log"
+
+	"github.com/BOPR/contracts/accountregistry"
+	"github.com/BOPR/contracts/create2transfer"
+	"github.com/BOPR/contracts/massmigration"
+	"github.com/BOPR/contracts/rollup"
+	"github.com/BOPR/contracts/state"
+	"github.com/BOPR/contracts/transfer"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	ethCmn "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// IBazooka is the common interface using which we will interact with the contracts
+// and the ethereum chain
+type IBazooka interface {
+	FetchBatchInputData(txHash ethCmn.Hash) (txs [][]byte, err error)
+}
+
+var (
+	ErrTxPending           = errors.New("Tx Pending, cannot read calldata")
+	ErrConvertingTxPayload = errors.New("Error converting tx payload")
+	ErrTxParamDoesntExist  = errors.New("Tx param does not exist")
+	ErrNoTxs               = errors.New("Error no transactions")
+	DOMAIN                 = [32]byte{}
+)
+
+const (
+	TXS_PARAM      = "txss"
+	META_PARAM     = "meta"
+	WITHDRAW_ROOTS = "withdrawRoots"
+)
+
+type (
+
+	// Bazooka contains everything needed to interact with smart contracts
+	Bazooka struct {
+		log       tmLog.Logger
+		EthClient *ethclient.Client
+
+		RollupABI abi.ABI
+		SC        Contracts
+	}
+
+	Contracts struct {
+		RollupContract  *rollup.Rollup
+		State           *state.State
+		Transfer        *transfer.Transfer
+		Create2Transfer *create2transfer.Create2transfer
+		MassMigration   *massmigration.Massmigration
+		AccountRegistry *accountregistry.Accountregistry
+	}
+)
+
+// NewPreLoadedBazooka creates
```

Comment:
nitpick: looks like you were about to document something but didn't finish for some reason.